File: azure-functions/csharp/Handlers/CustomerHandler.cs
using System.Linq;
using System.Threading.Tasks;
using Microsoft.AspNetCore.Mvc;
using Microsoft.Azure.WebJobs;
using Microsoft.Azure.WebJobs.Extensions.Http;
using Microsoft.AspNetCore.Http;
using Microsoft.Extensions.Logging;
using System.Collections.Generic;
using Company.Sales.Api.Models;
using System;

namespace Company.Sales.Api.Handlers
{
    public static class CustomerHandler
    {
        internal const string CollectionName = "Customers";

        [FunctionName("GetCustomers")]
        public static IActionResult GetCustomers(
            [HttpTrigger(AuthorizationLevel.Anonymous, "get", Route = "customers")] HttpRequest req,
            [CosmosDB(
                databaseName: "%DatabaseName%",
                collectionName: CollectionName,
                ConnectionStringSetting = "CosmosConnectionStr",
                SqlQuery = "SELECT * FROM c")] IEnumerable<Customer> customers,
            ILogger log)
        {
            log.LogInformation($"{customers.Count()} records retrieved.");
            return new OkObjectResult(customers);
        }

        [FunctionName("GetCustomer")]
        public static IActionResult GetCustomer(
            [HttpTrigger(AuthorizationLevel.Anonymous, "get", Route = "customers/{id:guid}")] HttpRequest req,
            [CosmosDB(
                databaseName: "%DatabaseName%",
                collectionName: CollectionName,
                ConnectionStringSetting = "CosmosConnectionStr",
                Id = "{id}",
                PartitionKey = "{id}")] Customer customer,
            Guid id,
            ILogger log)
        {
            if (customer != null)
            {
                return new OkObjectResult(customer);
            }
            else
            {
                return new BadRequestObjectResult($"No customer with Id {id}");
            }
        }

        [FunctionName("CreateCustomer")]
        public static async Task<IActionResult> CreateCustomer(
            [HttpTrigger(AuthorizationLevel.Anonymous, "post", Route = "customers")] Customer customer,
            [CosmosDB(
                databaseName: "%DatabaseName%",
                collectionName: CollectionName,
                ConnectionStringSetting = "CosmosConnectionStr")] IAsyncCollector<Customer> customerCollector,
            ILogger log)
        {
            customer.Id = Guid.NewGuid();
            await customerCollector.AddAsync(customer);
            return new CreatedResult($"/customers/{customer.Id}", customer);
        }
    }
}

File: azure-functions/java/src/main/java/com/company/sales/api/handlers/CustomerHandler.java
package com.company.sales.api.handlers;

import com.microsoft.azure.functions.*;
import com.microsoft.azure.functions.annotation.*;

import java.util.UUID;

import com.company.sales.api.models.*;

public class CustomerHandler {
    final static String CollectionName = "Customers";

    @FunctionName("GetCustomers")
    public HttpResponseMessage getCustomers(
            @HttpTrigger(
                name = "req",
                methods = {HttpMethod.GET},
                authLevel = AuthorizationLevel.ANONYMOUS,
                route = "customers") HttpRequestMessage<String> request,
            @CosmosDBInput(name = "database",
                databaseName = "%DatabaseName%",
                collectionName = CollectionName,
                sqlQuery = "select * from c",
                connectionStringSetting = "CosmosConnectionStr") Customer[] customers,
            final ExecutionContext context) {
        context.getLogger().info(customers.length + " records retrieved.");

        // Parse query parameter
        // final String query = request.getQueryParameters().get("name");
        // final String name = request.getBody().orElse(query);

        return request.createResponseBuilder(HttpStatus.OK).body(customers).build();
    }

    @FunctionName("GetCustomer")
    public HttpResponseMessage getCustomer(
            @HttpTrigger(
                name = "req",
                methods = {HttpMethod.GET},
                authLevel = AuthorizationLevel.ANONYMOUS,
                route = "customers/{id}") HttpRequestMessage<String> request,
            @CosmosDBInput(name = "database",
                databaseName = "%DatabaseName%",
                collectionName = CollectionName,
                id = "{id}",
                partitionKey = "{id}",
                connectionStringSetting = "CosmosConnectionStr") Customer customer,
            @BindingName("id") final UUID id,
            final ExecutionContext context) {
        if (customer != null) {
            return request.createResponseBuilder(HttpStatus.OK).body(customer).build();
        } else {
            return request.createResponseBuilder(HttpStatus.NOT_FOUND).body("No customer with Id " + id).build();
        }
    }

    @FunctionName("CreateCustomer")
    public HttpResponseMessage createCustomer(
            @HttpTrigger(
                name = "req",
                methods = {HttpMethod.POST},
                authLevel = AuthorizationLevel.ANONYMOUS,
                route = "customers") HttpRequestMessage<Customer> request,
            @CosmosDBOutput(name = "database",
                databaseName = "%DatabaseName%",
                collectionName = CollectionName,
                connectionStringSetting = "CosmosConnectionStr") OutputBinding<Customer> customerOutput,
            final ExecutionContext context) {
        Customer customer = request.getBody();
        customer.id = UUID.randomUUID();
        customerOutput.setValue(customer);

        return request.createResponseBuilder(HttpStatus.CREATED).body(customer).build();
    }
}
